internal/common/fwtypes/json_string.go

```diff
@@ -0,0 +1,141 @@
+package fwtypes
```

Comment:
This is to allow different JSON formats (whitespace/no-whitespace/line breaks etc.)
- See [this comment](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2519#discussion_r1727034943) for a full example.
- See [the TPF docs](https://developer.hashicorp.com/terraform/plugin/framework/migrating/resources/crud#planned-value-does-not-match-config-value) for the `Planned Value does not match Config Value` workaround.